### DISCUSSION TITLE
When and how should `nsw-layout` be used?

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [digitalnsw/nsw-design-system#220] [General]
The current documentation described the _Layout_ class having to be used in conjunction with a `nsw-container` to provide the correct spacing and alignment for content on the page. My initial understanding based on this description was that the entire page should be wrapped in a container and a layout (then either `__main` and/ or `__sidebar` used when needed). However, looking through the templates, the `nsw-layout` class gets used in other ways. On the [hero banner example](https://digitalnsw.github.io/nsw-design-system/templates/homepage/hero-banner.html), `nsw-layout` is used within each `nsw-section` to make further spacing adjustments. The next example of the `nsw-layout` is on the [search example](https://digitalnsw.github.io/nsw-design-system/templates/search/simple.html), where the entire page is contained within an `nsw-container` and an `nsw-layout`, whilst having an empty sidebar to limit the content width to 2 thirds of the page (on desktop). Finally, on the [filtering example](https://digitalnsw.github.io/nsw-design-system/templates/content/filters.html), a standalone `nsw-layout` is used on the introduction text to constrain the width and then a second instance is used to provide the filtering sidebar. Are all of the previously mentioned examples recommended ways of using the `nsw-layout`, or are they simply design decisions for each of the pages?

In short: is the `nsw-layout` class used to provide any visual meaning on the page/ match the NSW branding, or is it another spacing utility class?

**Comment by anna-yeshtukova** [comment]
Hi Jefferson,

Sorry for delayed reply.
Layout core component has been created to help DS users manage spacing and alignment. It's not a part of NSW branding requirements and doesn't have particular visual meaning but a useful tool to keep your page layouts consistent. 
In our examples we mostly use it when page requires LHS or RHS column. It has different spacing to what you can achieve out of the box with grid as it provides more padding between side column and main content. We highly recommend you to use it for cases such as filters or side navigation but for a full width content it's entirely up to you if you want to use it or not.

Hopefully this answers your question and please let us know if there's anything else that we can help you with.
